types/shapediver/appbuilderinteractiontypes.ts

```diff
@@ -0,0 +1,106 @@
+
+/**
+ * PART 1 - Parameter Definitions
+ * 
+ * These are the properties that are defined in GH.
+ * These definitions will in the future live in the Viewer.
+ */
+
+/**
+ * General properties of an interaction parameter.
+ */
+export interface IGeneralInteractionParameterProperties {
+    /** If the objects are hoverable. (default: true) */
+    hover?: boolean,
+    /** The names of the objects that can be interacted with. (see Jira document and discussion result) */
+    nameFilter?: string[]
+}
+
+/**
+ * Properties of a selection parameter.
+ */
+export interface IInteractionParameterPropsSelection extends IGeneralInteractionParameterProperties {
+    /** The minimum number of objects that can be selected. (default: 1) */
+    minimumSelection?: number,
+    /** The maximum number of objects that can be selected. (default: 1) */
+    maximumSelection?: number,
+}
+
+/**
+ * Properties of a dragging parameter.
+ * 
+ * This is a placeholder for now.
+ */
+export interface IInteractionParameterPropsDragging extends IGeneralInteractionParameterProperties {}
+
+/**
+ * Properties of a gumball parameter.
+ */
+export interface IInteractionParameterPropsGumball extends IGeneralInteractionParameterProperties {
+    /** The initial frame of the gumball. */
+    initialFrame?: any, // TODO: How to define this?
+    /** If the gumball can translate. (default: true) */
+    translation?: boolean,
+    /** If the gumball can rotate. (default: true) */
+    rotation?: boolean,
+    /** If the gumball can scale. (default: true) */
+    scaling?: boolean
+}
+
+/**
+ * The definition of an interaction parameter.
+ * 
+ * For each type, there is a corresponding set of properties.
+ */
+export interface IInteractionParameterDefinition {
+	/** Type of the interaction parameters. */
+	type: InteractionParameterType
+	/** Properties of the parameter definition. */
+	props: IInteractionParameterPropsSelection |
+        IInteractionParameterPropsDragging |
+        IInteractionParameterPropsGumball
+}
+
+
+/**
+ * PART 2 - Parameter Values
+ * 
+ * These are the definition of the JSON values that are sent from the Viewer.
+ * Although the actual type is a string (JSON stringified), we define the structure here.
+ */
+
+/**
+ * General properties of an interaction parameter value.
+ */
+export interface IGeneralInteractionProperties {
+    /** The names of the objects that are interacted with. */
+    names?: string[]
+}
+
+/**
+ * Properties of a transformation interaction value (gumball or dragging).
+ * 
+ * TODO: How to define the transformation values? Should this be a matrix?
+ */
+export interface ITransformationInteractionProperties extends IGeneralInteractionProperties {
+    /** The translation values. */
+    translation?: number[][], // TODO: How to define this?
+    /** The rotation values. */
+    rotation?: number[][], // TODO: How to define this?
+    /** The scaling values. */
+    scaling?: number[][], // TODO: How to define this?
```

Comment:
@MajorMeerkatThe3rd here we need to use a single property for storing the 4x4 transformation matrix. I am pretty sure one can't get separate values for translation/rotation/scaling from the Gumball, because these transformations are not commutative (the result depends on the order). Let's use the same property name and order of storing the matrix coefficients as defined by glTF.

Reply:
@snabela @MajorMeerkatThe3rd Agreed that these have to be matrices if we want to use the same objects for all types of inputs. However, it will be then likely be needed to do some sanity check / project of the matrix in case we know it is a pure transformation / rotation / scaling.

Reply:
@mathieu-shapediver there is a possibility to get Grasshopper to show this information, I checked it

Reply:
@snabela @mathieu-shapediver ok great, I'll change it to a single matrix within the next iteration. Much easier either way.